lib/membrane_live_web/channels/event_channel.ex

```diff
@@ -498,58 +476,43 @@ defmodule MembraneLiveWeb.EventChannel do
   end
 
   @impl true
-  def handle_in("mediaEvent", %{"data" => event}, socket) do
-    send(socket.assigns.event_pid, {:media_event, socket.assigns.peer_id, event})
+  def handle_in("mediaEvent", %{"data" => media_event}, socket) do
+    EventService.media_event(socket.assigns.event_id, socket.assigns.peer_id, media_event)
     {:noreply, socket}
   end
 
   @impl true
   def handle_in("isPlaylistPlayable", _data, socket) do
-    socket = set_event_pid(socket)
+    case EventService.playable_playlist(socket.assigns.event_id) do
+      {:error, _reason} ->
+        {:noreply, socket}
 
-    if socket.assigns.event_pid != nil do
-      case :global.whereis_name(socket.assigns.event_id) do
-        :undefined ->
-          {:reply, {:ok, false}, socket}
-
-        _pid ->
-          {:reply, {:ok, GenServer.call(socket.assigns.event_pid, :is_playlist_playable)}, socket}
-      end
-    else
-      {:reply, {:ok, false}, socket}
+      response ->
+        {:reply, {:ok, response}, socket}
     end
   end
 
   defp create_event(socket) do
-    case :global.whereis_name(socket.assigns.event_id) do
-      :undefined ->
-        maybe_start_event(socket)
+    room_exists? = EventService.room_exists?(socket.assigns.event_id)
 
-      pid ->
-        {:ok, pid}
-    end
-    |> case do
-      {:ok, event_pid} ->
-        {:ok, Socket.assign(socket, %{event_pid: event_pid})}
-
-      {:error, {:already_started, event_pid}} ->
-        {:ok, Socket.assign(socket, %{event_pid: event_pid})}
-
-      {:error, :non_moderator} ->
-        {:ok, Socket.assign(socket, %{event_pid: nil})}
-
-      {:error, reason} ->
+    case maybe_start_event(room_exists?, socket) do
+      {:error, :webinar_finished} ->
         Logger.error("""
         Failed to start room.
-        Room: #{inspect(socket.assigns.event_id)}
-        Reason: #{inspect(reason)}
+        Room: #{inspect(socket.assigns.event_id)}.
+        Reason: webinar is finished.
         """)
 
-        {:error, %{reason: "failed to start event: #{reason}"}}
+        {:error, %{reason: "failed to start event:  webinar is finished."}}
+
+      _else ->
+        {:ok, socket}
```

Comment:
Shouldn't we check if this else is not some other error?

Reply:
If you look at the function inside the case there are four return types:
```
:ok | :already_started | {:error, :webinar_finished} |  {:error, :non_moderator}
```
only in case of `{:error, :webinar_finished}` we want to return error.

Reply:
Then shouldn't we change it to if? Because the case with only two matches looks pretty strange.